core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCatalog.java

```diff
@@ -237,6 +239,24 @@ public void testDropTable() throws Exception {
     Assert.assertFalse(fs.isDirectory(new Path(metaLocation)));
   }
 
+  @Test
+  public void testDropNonIcebergTable() throws Exception {
+    Configuration conf = new Configuration();
+    String warehousePath = temp.newFolder().getAbsolutePath();
+    HadoopCatalog catalog = new HadoopCatalog(conf, warehousePath);
+    TableIdentifier testTable = TableIdentifier.of("db", "ns1", "ns2", "tbl");
+    String metaLocation = catalog.defaultWarehouseLocation(testTable);
+
+    FileSystem fs = Util.getFs(new Path(metaLocation), conf);
+    File tablePath = new File(warehousePath + "/db/ns1/ns2/tbl");
+    tablePath.mkdirs();
```

Comment:
This shouldn't mix Hadoop FileSystem calls with the Java File API. Please only use the Hadoop calls based on metadataLocation. That way you don't need to assume a location for the table.

Reply:
Makes sense, have made the change.